sysapps/common/common_api.js

```diff
@@ -80,13 +71,18 @@ var BindingObjectPrototype = function() {
     });
   };
 
-  function addMethodWithPromise(name, Promise) {
+  function addMethodWithPromise(name) {
     Object.defineProperty(this, name, {
       value: function() {
-        var promise_instance = new Promise();
         var args = Array.prototype.slice.call(arguments);
-        this._postMessage(name, args, wrapPromiseAsCallback(promise_instance));
-        return promise_instance;
+        return new Promise((resolve, reject) => {
+          this._postMessage(name, args, function(data, error_message) {
+            if (error_message)
+              reject(new Error(error_message));
```

Comment:
I did, see how I had to change `common_api_browsertest.html`. I'm using `Error` as described here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error